processor/resourcedetectionprocessor/internal/gcp/gcp.go

```diff
@@ -92,6 +92,25 @@ func (d *detector) Detect(context.Context) (resource pcommon.Resource, schemaURL
 		if d.resourceAttributes.CloudRegion.Enabled {
 			b.add(conventions.AttributeCloudRegion, d.detector.FaaSCloudRegion)
 		}
+	case gcp.CloudRunJob:
+		if d.resourceAttributes.CloudPlatform.Enabled {
+			b.attrs.PutStr(conventions.AttributeCloudPlatform, conventions.AttributeCloudPlatformGCPCloudRun)
```

Comment:
@jsuereth @dashpole @punya this is the only thing really blocking this, need to decide if Jobs fall under `AttributeCloudPlatformGCPCloudRun` or should be their own platform.

leaving this draft until that's resolved so it doesn't merge on accident